### ISSUE TITLE
$all support for Linq

### ISSUE THREAD

**Issue by oliverw** [craiggwilson/fluent-mongo#7]
It would be nice if $all would be supported using Linq.

**Comment by craiggwilson** [comment]
I began to implement this feature but after getting into it, figured out that this type of query isn't supported by mongodb.  In other words, there is no way to accomplish this.

Do you have any ideas on how it can be implemented?

**Comment by oliverw** [comment]
Hmm I was referring to this: 

http://www.mongodb.org/display/DOCS/Advanced+Queries#AdvancedQueries-%24all

**Comment by craiggwilson** [comment]
Yeah, I thought so.  That doesn't translate into the "All" linq method.  In fact, I don't think it translates into a linq method at all.  How do you propose this be supported?

**Comment by oliverw** [comment]
Hehe but I wasn't referring the Linq All() method but functionality equivalent to what the MongoDb $all operator does :)

**Comment by craiggwilson** [comment]
Yeah, I got that after you answered.

Still, my question remains, what syntax would you like to support this type of query?  I don't think there are any built-in operators that do this.

**Comment by oliverw** [comment]
I'm no expert in writing Linq Providers. Couldn't that go into an extension method in a driver specific class?

**Comment by craiggwilson** [comment]
Ok, so you're asking for this in a way that would translate to other linq providers.  I'll mark this as a feature request.
